controller/program/tests/tests/set_sol_value_calculator.rs

```diff
@@ -0,0 +1,705 @@
+                .with_is_disabled(true)
+                .with_is_rebalancing(false)
+                .build().0.map(|x| Some(Just(x).boxed()))),
+                ..Default::default()
+            }),
+            any_normal_pk(),
+            any::<u64>(),
+        ).prop_flat_map(
+                |(pool, mint_addr, spl_lamports)| (
+                    Just(pool),
+                    any_normal_pk().prop_filter("cannot be eq mint_addr", move |x| *x != mint_addr),
+                    any_spl_stake_pool(GenStakePoolArgs {
+                        pool_mint: Some(Just(mint_addr).boxed()),
+                        u64s: SplStakePoolU64s(NewSplStakePoolU64sBuilder::start()
+                            .with_last_update_epoch(Just(0).boxed()) // mollusk clock defaults to epoch 0
+                            .with_total_lamports(Just(spl_lamports).boxed())
+                            .with_pool_token_supply((spl_lamports / MAX_LAMPORTS_OVER_SUPPLY..=u64::MAX).boxed())
+                            .build().0.map(Some)),
+                        ..Default::default()
+                    }),
+                    any_lst_state(GenLstStateArgs {
+                        sol_value: Some((0..=pool.total_sol_value).boxed()),
+                        pks: LstStatePks(NewLstStatePksBuilder::start()
+                        .with_mint(mint_addr)
+                        .with_sol_value_calculator(sanctum_spl_multi::ID)
+                        .build().0.map(|x| Some(Just(x).boxed()))),
+                        ..Default::default()
+                    }),
+                )
+            ).prop_flat_map(
+                |(pool, stake_pool_addr, stake_pool, lsd)| (
+                    Just(pool),
+                    Just(lsd),
+                    Just(stake_pool_addr),
+                    Just(stake_pool),
+                    0..=max_sol_val_no_overflow(pool.total_sol_value, lsd.lst_state.sol_value) / MAX_LAMPORTS_OVER_SUPPLY,
+                )
+            ),
+        lsl in any_lst_state_list(Default::default(), 0..=MAX_LST_STATES),
+    ) {
+        set_sol_value_calculator_error_proptest(pool, lsl, lsd, stake_pool_addr, stake_pool, new_balance, pool.admin, TestErrorType::PoolDisabled).unwrap();
+    }
+}
+
+fn set_sol_value_calculator_wsol_proptest(
+    pool: PoolState,
+    mut lsl: LstStateListData,
+    wsol_lsd: LstStateData,
+    new_balance: u64,
+) -> TestCaseResult {
+    silence_mollusk_logs();
+    let lst_idx = lsl.upsert(wsol_lsd);
+    let LstStateListData {
+        lst_state_list,
+        all_pool_reserves,
+        ..
+    } = lsl;
+
+    // Set initial calculator to a random pubkey
+    let mut lsl_data = lst_state_list.clone();
+    let lsl_mut = LstStatePackedListMut::of_acc_data(&mut lsl_data).unwrap();
+    let lst_mut = unsafe { lsl_mut.0.get_mut(lst_idx).unwrap().as_lst_state_mut() };
+    lst_mut.sol_value_calculator = Pubkey::new_unique().to_bytes();
```

Comment:
since this meant to be used in a proptest, this should be passed in as an argument so that it can be randomly generated by a proptest strategy